File: problems/0283-move-zeroes/solution.c
void moveZeroes(int* nums, int numsSize) {
    int i = 0;
    
    while (i < numsSize) {
        if (nums[i] == 0) {
            for (int j = i; j < numsSize - 1; j++) {
                nums[j] = nums[j + 1];
            }
            nums[numsSize - 1] = 0;
            numsSize--; 
        } else {
            i++;  
        }
    }
}

File: problems/0283-move-zeroes/solution.java
class Solution {
    public void moveZeroes(int[] nums) {
        int zeroCount=0;

        for(int num:nums){
            if(num!=0){
                nums[zeroCount++]=num;
            }
        }

        while(zeroCount<nums.length){
            nums[zeroCount++]=0;
        }
    }
}
